nemo/collections/asr/metrics/wer.py

```diff
@@ -270,32 +341,43 @@ def decode_hypothesis(
 
                 # CTC decoding procedure
                 decoded_prediction = []
+                token_lengths = []  # preserve token lengths
                 token_repetitions = []  # preserve number of repetitions per token
 
                 previous = self.blank_id
-                last_repetition = 0
+                last_length = 0
+                last_repetition = 1
 
                 for pidx, p in enumerate(prediction):
                     if (p != previous or previous == self.blank_id) and p != self.blank_id:
                         decoded_prediction.append(p)
 
-                        token_repetitions.append(pidx - last_repetition)
-                        last_repetition = pidx
+                        token_lengths.append(pidx - last_length)
+                        last_length = pidx
+                        token_repetitions.append(last_repetition)
+                        last_repetition = 1
+
+                    if p == previous and previous != self.blank_id:
+                        last_repetition += 1
 
                     previous = p
 
+                if len(token_repetitions) > 0:
+                    token_repetitions = token_repetitions[1:] + [last_repetition]
+
             else:
                 if predictions_len is not None:
                     prediction = prediction[:predictions_len]
                 decoded_prediction = prediction[prediction != self.blank_id].tolist()
+                token_lengths = [1] * len(decoded_prediction)  # preserve number of repetitions per token
                 token_repetitions = [1] * len(decoded_prediction)  # preserve number of repetitions per token
 
             # De-tokenize the integer tokens; if not computing timestamps
             if self.compute_timestamps is True:
                 # keep the original predictions, wrap with the number of repetitions per token
                 # this is done so that `ctc_decoder_predictions_tensor()` can process this hypothesis
                 # in order to compute exact time stamps.
-                hypothesis = (decoded_prediction, token_repetitions)
+                hypothesis = (decoded_prediction, token_lengths, token_repetitions)
```

Comment:
You have updated this, but have not updated signature below. This will cause breakage in CTC word timestamp calculation.